### PR TITLE
[FIX] Create frontend and connect backend to Google Calendar API

### PR DESCRIPTION
**Pull request by pzmarzly**
Connect to Google APIs and create frontend.

Follow up to #18 (it was not ready to be merged).

### CODE REVIEW

File: frontend/src/components/CalendarList/CalendarListItem/CalendarListItem.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+const CalendarListItem = ( { calendarData } ) => {
+   const { description, summary,id } = calendarData;
+
+   return ( 
+      <StyledCalendarListItem>
+         <CalendarLink to={ `/room/${id.split( '@' )[0]}` }>
+
+            <CalendarHeader >{ summary }</CalendarHeader>
+            <CalendarDescription >
```
Comment (MaciejChyrc) [inline frontend/src/components/CalendarList/CalendarListItem/CalendarListItem.js]: Unnecessary space in opening tag. Shouldn't your linter catch it?

File: frontend/src/pages/Calendars/Calendars.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      
+      fetch( `${constants.url.API_URL}calendars`,{ signal } )
+         .then( ( response ) => response.json() )
+         .then( ( data ) => {
+            const calendarList = data.filter(
+               ( calendar ) => calendar.summary.slice( 0, 5 ) === 'ROOM_',
+            );
+            setCalendars( calendarList );
+            setIsLoading( false );
+         } );
```
Comment (MaciejChyrc) [inline frontend/src/pages/Calendars/Calendars.js]: What if fetching data fails? There should be a `catch` function to handle promise reject. Without it I assume you'd have loading text infinitely without giving a user any info that data isn't actually loading but failed to load. Same for `Room` file below.

File: frontend/src/pages/Calendars/Calendars.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react';
+import PageTemplate from '../../templates/PageTemplate/PageTemplate';
+import CalendarList from '../../components/CalendarList/CalendarList';
+import { constants } from '../../assets/configs/constants';
+
+const Calendars = () => {
+   const [ calendars, setCalendars ] = useState( [] );
+   const [ isLoading, setIsLoading ] = useState( true );
+   useEffect( () => {
+      const controller = new AbortController();
```
Comment (MaciejChyrc) [inline frontend/src/pages/Calendars/Calendars.js]: Where does `AbortController` come from and what does it do? I can't see it among imports.

File: frontend/src/templates/PageTemplate/PageTemplate.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import PropTypes from 'prop-types';
+import { PageWrapper } from './PageTemplate_styles';
+import GlobalStyle from '../../assets/styles/GlobalStyle';
+
+import { theme } from '../../assets/styles/theme';
+import NavBar from '../../components/NavBar/NavBar';
+
+const PageTemplate = ( { children } ) => {
+   return (
+      <ThemeProvider theme={ theme }>
```
Comment (MaciejChyrc) [inline frontend/src/templates/PageTemplate/PageTemplate.js]: You could move `ThemeProvider`, global page layout (only if there's one layout used in all your views) and other providers if there will be more to the `App` component. They'd be loaded once then on application start.

File: frontend/src/pages/Calendars/Calendars.js (hunk 3 above)
Comment (oleks12345) [reply frontend/src/pages/Calendars/Calendars.js]: https://developer.mozilla.org/en-US/docs/Web/API/AbortController

File: backend/app/calendar.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+const { GoogleCalendarMock } = require( './google-calendar-mock' );
+
+let calendar;
+
+if ( typeof jest === 'undefined' ) {
+   calendar = google.calendar( { version: 'v3', auth: authorize() } );
+}
+else {
+   calendar = new GoogleCalendarMock();
+}
```
[the remark this answers is not in the document]
Comment (pzmarzly) [reply backend/app/calendar.js]: Moved to #22 , should be fixed by #23 .

**Pull request merged**